aten/src/ATen/native/Pow.cpp

```diff
@@ -6,85 +6,128 @@
+
+  //if (exp.equal(0.0)) {
+    //resize_output(result, base.sizes());
+    //result.fill_(1);
+    //namedinference::propagate_names(result, base);
+  //} else if (exp.equal(1.0)) {
+    //resize_output(result, base.sizes());
+    //result.copy_(base);
+    //namedinference::propagate_names(result, base);
+  //} else {
+    //auto iter = TensorIterator::unary_op(result, base.to(common_dtype));
+    //pow_tensor_scalar_stub(iter.device_type(), iter, exp);
+  //}
+  //return result;
+//}
+
+Tensor& pow_out(Scalar base, const Tensor& exp, Tensor& result) {
+  auto& result_ = const_cast<Tensor&>(result);
   if (base.isComplex() && base.toComplexDouble() == 1.0) {
-    resize_output(result, exp.sizes());
-    result.fill_(1);
-    namedinference::propagate_names(result, exp);
+    resize_output(result_, exp.sizes());
+    result_.fill_(1);
+    namedinference::propagate_names(result_, exp);
   } else if (!base.isComplex() && base.toDouble() == 1.0) {
-    resize_output(result, exp.sizes());
-    result.fill_(1);
-    namedinference::propagate_names(result, exp);
+    resize_output(result_, exp.sizes());
+    result_.fill_(1);
+    namedinference::propagate_names(result_, exp);
   } else {
-    native::pow_out(result, c10::scalar_to_tensor(base, exp.device()), exp);
+    at::pow_out(result_, c10::scalar_to_tensor(base, exp.device()), exp); // redispatch!
   }
-  return result;
+  return result_;
 }
 
-Tensor& pow_(Tensor& base, const Tensor& other) {
-  return native::pow_out(base, base, other);
+Tensor& pow_(const Tensor& base, const Tensor& exp) {
+  return at::pow_out(const_cast<Tensor&>(base), base, exp); // redispatch!
 }
 
-Tensor& pow_(Tensor& base, Scalar alpha) {
-  return native::pow_out(base, base, alpha);
+Tensor& pow_(const Tensor& base, const Scalar exp) {
+  return at::pow_out(const_cast<Tensor&>(base), base, exp); // redispatch!
 }
 
-Tensor pow(const Tensor& base, const Tensor& exp) {
-  auto dtype = at::result_type(base, exp);
-  Tensor result = at::empty({0}, base.options().dtype(dtype));
-  return native::pow_out(result, base, exp);
-}
+//Tensor pow(const Tensor& base, const Tensor& exp) {
+  //auto dtype = at::result_type(base, exp);
+  //Tensor result = at::empty({0}, base.options().dtype(dtype));
+  //return native::pow_out(result, base, exp);
+//}
 
-Tensor pow(const Tensor& base, Scalar exp) {
-  auto dtype = at::result_type(base, exp);
-  Tensor result = at::empty_like(base, base.options().dtype(dtype), MemoryFormat::Preserve);
-  return native::pow_out(result, base, exp);
-}
+//Tensor pow(const Tensor& base, Scalar exp) {
+  //auto dtype = at::result_type(base, exp);
+  //Tensor result = at::empty_like(base, base.options().dtype(dtype), MemoryFormat::Preserve);
+  //return native::pow_out(result, base, exp);
+//}
 
 Tensor pow(Scalar base, const Tensor& exp) {
-  auto dtype = at::result_type(base, exp);
-  Tensor result = at::empty_like(exp, exp.options().dtype(dtype), MemoryFormat::Preserve);
```

Comment:
I'm losing the `MemoryFormat::Preserve` here, since I can't forward that information to the overload call. I think that's fine, since (a) it looks like `empty_like` [defaults](https://github.com/pytorch/pytorch/blob/1c9fc38eb2bc687487471e05ea20cd3e2359f538/aten/src/ATen/native/TensorFactories.cpp#L251) to `Memory_format::Preserve` anyway, and (b) I'm relying on passing tests.